core/src/libraries/BalanceSheet.sol

```diff
@@ -84,16 +93,16 @@ library BalanceSheet {
         uint256 liabilities0,
         uint256 liabilities1,
         uint160 sqrtMeanPriceX96
-    ) internal pure returns (uint256 incentive1) {
+    ) internal pure returns (uint256 incentive1, uint224 meanPriceX96) {
         unchecked {
-            uint256 meanPriceX96 = Math.mulDiv(sqrtMeanPriceX96, sqrtMeanPriceX96, Q96);
+            meanPriceX96 = uint224(Math.mulDiv(sqrtMeanPriceX96, sqrtMeanPriceX96, Q96));
 
             if (liabilities0 > assets0) {
                 // shortfall is the amount that cannot be directly repaid using Borrower assets at this price
                 uint256 shortfall = liabilities0 - assets0;
                 // to cover it, a liquidator may have to use their own assets, taking on inventory risk.
                 // to compensate them for this risk, they're allowed to seize some of the surplus asset.
-                incentive1 += Math.mulDiv(shortfall, meanPriceX96, Q96) / LIQUIDATION_INCENTIVE;
+                incentive1 += Math.mulDiv(shortfall / LIQUIDATION_INCENTIVE, meanPriceX96, Q96);
```

Comment:
Equivalent expressions 👍

Reply:
🙃

Reply:
There was an actual reason for this though. `shortfall` is _at most_ equal to `liabilities0`, and because of constraints in Lender, the maximum value of liabilities is approximately 2^184. The maximum value of `meanPriceX96` is 2^224-1, and `Q96` is 2^96. The result of `Math.mulDiv` must fit in 256 bits, which means if `liabilities0 > 2**128` this could fail. By dividing by `LIQUIDATION_INCENTIVE` _before_ entering the `mulDiv` step, I'm giving us an extra `log2(20)=4.32` bits of wiggle room.

Is this completely esoteric? Yes. Will token amounts > 2**128 break other things first? Almost certainly. But who doesn't love a 4.32 bit improvement?